### PR TITLE
use dto in responses

### PR DESCRIPTION
**Pull request by hosseyn-alavi**

### CODE REVIEW

File: src/tasks/tasks.controller.ts [hunk 1]
```diff
@@ -25,34 +25,51 @@ import { TasksService } from './tasks.service';
...
-  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  @ApiBearerAuth('accessToken')
+  @ApiResponse({
+    status: 201,
+    description: 'Task created',
+    type: CreateTaskReqDto,
+  })
   createTask(
-    @Body() createTaskDto: CreateTaskDto,
+    @Body() CreateTaskReqDto: CreateTaskReqDto,
```
Comment (a-molaei) [inline src/tasks/tasks.controller.ts:63]: variable better to be camelCase

File: src/tasks/tasks.controller.ts [hunk 2]
```diff
@@ -25,34 +25,51 @@ import { TasksService } from './tasks.service';
...
+  ): Promise<GetTaskResDto> {
     return this.tasksService.getTaskById(id, user);
   }
 
   @Post()
-  @ApiOperation({ summary: 'Create task' })
-  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  @ApiBearerAuth('accessToken')
+  @ApiResponse({
+    status: 201,
```
Comment (a-molaei) [inline src/tasks/tasks.controller.ts:58]: ApiCreatedResponse could be used

File: src/organisation/organisation.controller.ts [hunk 3]
```diff
@@ -5,14 +5,18 @@ import { GetUser } from 'src/auth/get-user.decorator';
...
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
 
 @UseGuards(AuthGuard())
 @ApiTags('Organisations')
 @Controller('organisation')
 export class OrganisationController {
   constructor(private organisationService: OrganisationService) {}
 
+  @ApiResponse({
+    status: 201,
```
Comment (a-molaei) [inline src/organisation/organisation.controller.ts:17]: ApiCreatedReponse could be used.

File: src/tasks/tasks.controller.ts [hunk 4]
```diff
@@ -25,34 +25,51 @@ import { TasksService } from './tasks.service';
 export class TasksController {
   constructor(private tasksService: TasksService) {}
 
+  @ApiResponse({
+    status: 200,
```
Comment (a-molaei) [inline src/tasks/tasks.controller.ts:29]: ApiOkResponseCould be used.

File: src/tasks/tasks.controller.ts [hunk 5]
```diff
@@ -25,34 +25,51 @@ import { TasksService } from './tasks.service';
...
     @GetUser() user: User,
-  ): Promise<Task> {
-    return this.tasksService.createTask(createTaskDto, user);
+  ): Promise<CreateTaskResDto> {
+    return this.tasksService.createTask(CreateTaskReqDto, user);
   }
 
+  @ApiBearerAuth('accessToken')
+  @ApiResponse({
+    status: 200,
```
Comment (a-molaei) [inline src/tasks/tasks.controller.ts:71]: ApiOkResponseCould be used.

File: src/tasks/tasks.controller.ts [hunk 6]
```diff
@@ -61,12 +78,16 @@ export class TasksController {
     return this.tasksService.deleteTaskById(id, user);
   }
 
+  @ApiBearerAuth('accessToken')
+  @ApiResponse({
+    status: 200,
```
Comment (a-molaei) [inline src/tasks/tasks.controller.ts:83]: ApiOkResponseCould be used.

File: src/tasks/tasks.controller.ts [hunk 7]
```diff
@@ -25,34 +25,51 @@ import { TasksService } from './tasks.service';
...
     @Query() filterDto: GetTasksFilterDto,
     @GetUser() user: User,
-  ): Promise<Task[]> {
+  ): Promise<GetTaskResDto[]> {
     return this.tasksService.getTasks(filterDto, user);
   }
 
   @Get('/:id')
   @ApiResponse({
     status: 200,
```
Comment (a-molaei) [inline src/tasks/tasks.controller.ts:43]: ApiOkResponseCould be used.

File: src/tasks/tasks.service.ts [hunk 8]
```diff
@@ -39,7 +42,7 @@ export class TasksService {
     id: string,
     status: TaskStatus,
     user: User,
-  ): Promise<Task> {
+  ): Promise<GetTaskResDto> {
```
Comment (a-molaei) [inline src/tasks/tasks.service.ts:45]: It doesn't make any difference to only change the return type of the function. Inside the function, a new object with the type of GetTaskResDto must be created and the entity fields must be copied into that object's fields.

File: src/tasks/tasks.service.ts [hunk 9]
```diff
@@ -1,31 +1,34 @@
...
       throw new NotFoundException();
     }
     return found;
   }
 
-  createTask(createTaskDto: CreateTaskDto, user: User): Promise<Task> {
+  createTask(
+    createTaskDto: CreateTaskReqDto,
+    user: User,
+  ): Promise<CreateTaskResDto> {
```
Comment (a-molaei) [inline src/tasks/tasks.service.ts:31]: Same here.

File: src/tasks/tasks.service.ts [hunk 10]
```diff
@@ -1,31 +1,34 @@
...
-import { User } from 'src/auth/user.entity';
 
 @Injectable()
 export class TasksService {
   constructor(
     @InjectRepository(TaskRepository) private taskRepository: TaskRepository,
   ) {}
 
-  getTasks(filterDto: GetTasksFilterDto, user: User): Promise<Task[]> {
+  getTasks(filterDto: GetTasksFilterDto, user: User): Promise<GetTaskResDto[]> {
```
Comment (a-molaei) [inline src/tasks/tasks.service.ts:16]: and here.

File: src/tasks/tasks.service.ts [hunk 11]
```diff
@@ -1,31 +1,34 @@
...
     return this.taskRepository.getTasks(filterDto, user);
   }
 
-  async getTaskById(id: string, user: User): Promise<Task> {
+  async getTaskById(id: string, user: User) {
```
Comment (a-molaei) [inline src/tasks/tasks.service.ts:20]: and here, besides that this function is missing the return type.

File: src/tasks/tasks.service.ts [hunk 12]
```diff
@@ -1,31 +1,34 @@
...
     const found = await this.taskRepository.findOne({ where: { id, user } });
```
Comment (a-molaei) [inline src/tasks/tasks.service.ts:21]: I think `task` could be a better variable name here.

File: src/tasks/tasks.controller.ts (hunk 1 above)
Comment (hosseyn-alavi) [reply src/tasks/tasks.controller.ts:63]: Oh , typo,
Thanks for the notice

File: src/tasks/tasks.service.ts (hunk 8 above)
Comment (hosseyn-alavi) [reply src/tasks/tasks.service.ts:45]: Just for the sake of using typescript and consistency

**Pull request merged**